src/main/java/knightminer/simplytea/worldgen/TeaTreeFeature.java

```diff
@@ -25,7 +27,7 @@ public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context) {
 		WorldGenLevel world = context.level();
 		BlockState soil = world.getBlockState(down);
 		RandomSource random = context.random();
-		if (world.getBlockState(pos).isAir()) {
+		if (IntStream.range(0, 4).allMatch(i -> world.isEmptyBlock(pos.above(i)))) {
```

Comment:
This is incorrect, the tree will never be 4 tall without a later check, 3 blocks above should be sufficient